networkx/readwrite/gexf.py

```diff
@@ -602,105 +579,107 @@ def indent(self, elem, level=0):
+        if edgedefault == 'directed':
+            G = nx.MultiDiGraph()
         else:
-            G=nx.MultiGraph()
+            G = nx.MultiGraph()
 
         # graph attributes
-        graph_start=graph_xml.get('start')
+        graph_name = graph_xml.get('name', None)
+        if graph_name is not None:
+            G.graph['name'] = graph_name
+        graph_start = graph_xml.get('start')
         if graph_start is not None:
-            G.graph['start']=graph_start
-        graph_end=graph_xml.get('end')
+            G.graph['start'] = graph_start
+        graph_end = graph_xml.get('end')
         if graph_end is not None:
-            G.graph['end']=graph_end
-        graph_mode=graph_xml.get("mode", "")
-        if graph_mode=='dynamic':
-            G.graph['mode']='dynamic'
+            G.graph['end'] = graph_end
+        graph_mode = graph_xml.get('mode', '')
+        if graph_mode == 'dynamic':
+            G.graph['mode'] = 'dynamic'
         else:
-            G.graph['mode']='static'
+            G.graph['mode'] = 'static'
 
         # timeformat
-        self.timeformat=graph_xml.get('timeformat')
+        self.timeformat = graph_xml.get('timeformat')
         if self.timeformat == 'date':
             self.timeformat = 'string'
 
         # node and edge attributes
-        attributes_elements=graph_xml.findall("{%s}attributes"%self.NS_GEXF)
+        attributes_elements = graph_xml.findall('{%s}attributes' % self.NS_GEXF)
         # dictionaries to hold attributes and attribute defaults
-        node_attr={}
-        node_default={}
-        edge_attr={}
-        edge_default={}
+        node_attr = {}
+        node_default = {}
+        edge_attr = {}
+        edge_default = {}
         for a in attributes_elements:
-            attr_class = a.get("class")
-            if attr_class=='node':
+            attr_class = a.get('class')
+            if attr_class == 'node':
                 na,nd = self.find_gexf_attributes(a)
                 node_attr.update(na)
                 node_default.update(nd)
-                G.graph['node_default']=node_default
-            elif attr_class=='edge':
+                G.graph['node_default'] = node_default
+            elif attr_class == 'edge':
                 ea,ed = self.find_gexf_attributes(a)
                 edge_attr.update(ea)
                 edge_default.update(ed)
-                G.graph['edge_default']=edge_default
+                G.graph['edge_default'] = edge_default
             else:
                 raise # unknown attribute class
 
         # Hack to handle Gephi0.7beta bug
         # add weight attribute
-        ea={'weight':{'type': 'double', 'mode': 'static', 'title': 'weight'}}
-        ed={}
+        ea = {'weight': {'type': 'double', 'mode': 'static', 'title': 'weight'}}
+        ed = {}
         edge_attr.update(ea)
         edge_default.update(ed)
-        G.graph['edge_default']=edge_default
+        G.graph['edge_default'] = edge_default
 
```

Comment:
I'm not sure if we still need this. Gephi currently is at 0.9.1. If I see it correctly, this hack creates an empty attribute named `edge_default`. What's your opinion about this?